javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,67 @@
+const products = getAvailableProducts();
+// console.log(products);
+
+const searchInput = document.querySelector("#searchInput");
+const maxPriceInput = document.querySelector("#maxPrice");
+const productsList = document.querySelector("ul");
+
+function renderProducts(products) {
+  for (let i = 0; i < products.length; i++) {
+    const title = document.createElement("li");
+    title.innerHTML = products[i].name;
+    title.style.fontWeight = "bold";
+    productsList.appendChild(title);
+
+    const price = document.createElement("li");
+    price.innerHTML = "price: " + products[i].price;
+    productsList.appendChild(price);
+
+    const rating = document.createElement("li");
+    rating.innerHTML = "rating: " + products[i].rating;
+    productsList.appendChild(rating);
+  }
+}
+
+renderProducts(products); // This should create the ul and the li's with the individual products details
+
+// console.log(products);
+// console.log(products[0]);
+
+// console.log(productsList);
+
+////////////////////////////// PART 2
+
+// 2.2 - Searching for products
+
+searchInput.addEventListener("keyup", function () {
+  const inputValue = searchInput.value;
+  const filteredProducts = products.filter((product) =>
+    product.name.toLowerCase().includes(inputValue.toLowerCase())
+  );
+  if (!inputValue || inputValue === " ") {
+    productsList.innerHTML = "";
+    alert("Please enter product name!");
+    // renderProducts(products);
```

Comment:
```suggestion
    renderProducts(products);
```
If you comment on this, the list of products is gone when you erase the input. The same happens on maxPrice.